src/util/path.js

```diff
@@ -465,6 +465,28 @@
     return tmpLen;
   }
 
+  // add explanation in docs
+  function findPercentageForDistance(segInfo, distance) {
+    var perc = 0, tmpLen = 0, iterator = segInfo.iterator, tempP = { x: segInfo.x, y: segInfo.y },
+        p, nextLen, nextStep = 0.01;
```

Comment:
I just did some quick performance testing. I get about 3x better performance by using an initial nextStep value of 0.1 rather than 0.01

Reply:
do you have some flamegraph to see where the time is wasted on? the problem with 0.1 is that with strong curves, small radius, may position characters incorrectly.

Reply:
I've never used a flamegraph before, but here are some performance logs for comparing execution time and total steps required for both 0.01 and 0.1. Longer distances obviously result in a large performance improvement, although I haven't tested with many types of curves to see where positioning issues might occur.

**initialStep: 0.01**
> Found 14.343359375 in 27 steps Time: 0.039999838918447495
Found 14.543359375 in 25 steps Time: 0.0300002284348011
Found 38.78671875 in 23 steps Time: 0.019999686628580093
Found 38.98671875 in 24 steps Time: 0.014999881386756897
Found 57.663671875 in 27 steps Time: 0.0200001522898674
Found 57.863671875 in 29 steps Time: 0.039999838918447495
Found 71.540625 in 30 steps Time: 0.019999686628580093
Found 71.740625 in 28 steps Time: 0.0350000336766243
Found 89.88046875 in 31 steps Time: 0.039999838918447495
Found 90.08046875 in 26 steps Time: 0.0200001522898674
Found 108.776953125 in 28 steps Time: 0.019999686628580093
Found 108.976953125 in 33 steps Time: 0.0700000673532486
Found 124.33359375 in 30 steps Time: 0.025000423192977905
Found 124.53359375 in 32 steps Time: 0.0350000336766243
Found 144.33359375 in 32 steps Time: 0.0700000673532486
Found 144.53359375 in 32 steps Time: 0.049999915063381195
Found 159.33359375 in 32 steps Time: 0.0300002284348011
Found 159.53359375 in 32 steps Time: 0.045000109821558
Found 169.890234375 in 31 steps Time: 0.0300002284348011
Found 170.090234375 in 34 steps Time: 0.059999991208314896
Found 185.446875 in 34 steps Time: 0.024999957531690598
Found 185.646875 in 37 steps Time: 0.049999915063381195
Found 200.446875 in 34 steps Time: 0.059999991208314896
Found 200.646875 in 35 steps Time: 0.05499972030520439
Found 212.10703125 in 38 steps Time: 0.029999762773513794
Found 212.30703125 in 37 steps Time: 0.04499964416027069
Found 228.7671875 in 37 steps Time: 0.049999915063381195
Found 228.9671875 in 38 steps Time: 0.0300002284348011
Found 244.323828125 in 35 steps Time: 0.039999838918447495
Found 244.523828125 in 40 steps Time: 0.0800001434981823
Found 258.757421875 in 39 steps Time: 0.039999838918447495
Found 258.957421875 in 40 steps Time: 0.08999975398182869
Found 272.634375 in 39 steps Time: 0.049999915063381195
Found 272.834375 in 40 steps Time: 0.0650002621114254
Found 283.191015625 in 44 steps Time: 0.0300002284348011
Found 283.391015625 in 41 steps Time: 0.0350000336766243
Found 298.74765625 in 41 steps Time: 0.0300002284348011
Found 298.94765625 in 41 steps Time: 0.039999838918447495
Found 317.624609375 in 43 steps Time: 0.045000109821558
Found 317.824609375 in 44 steps Time: 0.059999991208314896
Found 342.058203125 in 49 steps Time: 0.0400003045797348
Found 342.258203125 in 46 steps Time: 0.045000109821558
Found 362.61484375 in 50 steps Time: 0.0750003382563591
Found 362.81484375 in 52 steps Time: 0.0550001859664917
Found 376.491796875 in 51 steps Time: 0.039999838918447495
Found 376.691796875 in 49 steps Time: 0.059999991208314896
Found 390.925390625 in 51 steps Time: 0.039999838918447495
Found 391.125390625 in 54 steps Time: 0.0849999487400055
Found 402.038671875 in 55 steps Time: 0.049999915063381195
Found 402.238671875 in 56 steps Time: 0.049999915063381195
Found 412.5953125 in 54 steps Time: 0.0550001859664917
Found 412.7953125 in 58 steps Time: 0.059999991208314896
Found 422.5953125 in 65 steps Time: 0.0500003807246685
Found 422.7953125 in 57 steps Time: 0.06499979645013809
Found 437.5953125 in 65 steps Time: 0.08999975398182869
Found 437.7953125 in 63 steps Time: 0.1050001010298729
Found 457.5953125 in 68 steps Time: 0.0550001859664917
Found 457.7953125 in 67 steps Time: 0.06499979645013809
Found 476.472265625 in 73 steps Time: 0.06499979645013809
Found 476.672265625 in 71 steps Time: 0.3499998711049557
Found 490.34921875 in 79 steps Time: 0.3349999897181988
Found 490.54921875 in 77 steps Time: 0.08999975398182869
Found 502.009375 in 79 steps Time: 0.029999762773513794
Found 502.209375 in 81 steps Time: 0.049999915063381195
Found 514.226171875 in 81 steps Time: 0.014999881386756897
Found 514.426171875 in 81 steps Time: 0.045000109821558
Found 529.7828125 in 88 steps Time: 0.0200001522898674
Found 529.9828125 in 90 steps Time: 0.07499987259507179
Found 549.7828125 in 91 steps Time: 0.015000347048044205
Found 549.9828125 in 89 steps Time: 0.0350000336766243
Found 564.7828125 in 97 steps Time: 0.029999762773513794
Found 564.9828125 in 89 steps Time: 0.0650002621114254
Found 575.339453125 in 93 steps Time: 0.029999762773513794
Found 575.539453125 in 97 steps Time: 0.0300002284348011
Found 590.89609375 in 97 steps Time: 0.019999686628580093
Found 591.09609375 in 100 steps Time: 0.0700000673532486
Found 605.89609375 in 99 steps Time: 0.024999957531690598
Found 606.09609375 in 97 steps Time: 0.0300002284348011
Found 617.55625 in 105 steps Time: 0.019999686628580093
Found 617.75625 in 103 steps Time: 0.015000347048044205
Found 629.773046875 in 109 steps Time: 0.049999915063381195
Found 629.973046875 in 103 steps Time: 0.0200001522898674
Found 645.3296875 in 113 steps Time: 0.024999957531690598
Found 645.5296875 in 111 steps Time: 0.029999762773513794
Found 665.3296875 in 115 steps Time: 0.0200001522898674
Found 665.5296875 in 114 steps Time: 0

**initialStep: 0.1**
> Found 14.343359375 in 27 steps Time: 0.004999805241823196
Found 14.543359375 in 28 steps Time: 0
Found 38.78671875 in 28 steps Time: 0.004999805241823196
Found 38.98671875 in 28 steps Time: 0.004999805241823196
Found 57.663671875 in 31 steps Time: 0
Found 57.863671875 in 27 steps Time: 0
Found 71.540625 in 30 steps Time: 0
Found 71.740625 in 32 steps Time: 0.014999881386756897
Found 89.88046875 in 29 steps Time: 0.004999805241823196
Found 90.08046875 in 33 steps Time: 0
Found 108.776953125 in 29 steps Time: 0.004999805241823196
Found 108.976953125 in 31 steps Time: 0.005000270903110504
Found 124.33359375 in 32 steps Time: 0
Found 124.53359375 in 29 steps Time: 0.0100000761449337
Found 144.33359375 in 32 steps Time: 0.005000270903110504
Found 144.53359375 in 34 steps Time: 0
Found 159.33359375 in 32 steps Time: 0
Found 159.53359375 in 31 steps Time: 0
Found 169.890234375 in 31 steps Time: 0
Found 170.090234375 in 33 steps Time: 0
Found 185.446875 in 32 steps Time: 0.005000270903110504
Found 185.646875 in 34 steps Time: 0.005000270903110504
Found 200.446875 in 33 steps Time: 0.005000270903110504
Found 200.646875 in 32 steps Time: 0.004999805241823196
Found 212.10703125 in 29 steps Time: 0.005000270903110504
Found 212.30703125 in 28 steps Time: 0.019999686628580093
Found 228.7671875 in 31 steps Time: 0
Found 228.9671875 in 35 steps Time: 0.004999805241823196
Found 244.323828125 in 33 steps Time: 0.005000270903110504
Found 244.523828125 in 30 steps Time: 0.004999805241823196
Found 258.757421875 in 30 steps Time: 0.004999805241823196
Found 258.957421875 in 32 steps Time: 0.004999805241823196
Found 272.634375 in 27 steps Time: 0.005000270903110504
Found 272.834375 in 30 steps Time: 0.0200001522898674
Found 283.191015625 in 31 steps Time: 0
Found 283.391015625 in 31 steps Time: 0.014999881386756897
Found 298.74765625 in 30 steps Time: 0.004999805241823196
Found 298.94765625 in 29 steps Time: 0
Found 317.624609375 in 32 steps Time: 0.004999805241823196
Found 317.824609375 in 33 steps Time: 0
Found 342.058203125 in 33 steps Time: 0.004999805241823196
Found 342.258203125 in 28 steps Time: 0
Found 362.61484375 in 36 steps Time: 0
Found 362.81484375 in 28 steps Time: 0
Found 376.491796875 in 35 steps Time: 0.004999805241823196
Found 376.691796875 in 34 steps Time: 0
Found 390.925390625 in 31 steps Time: 0
Found 391.125390625 in 31 steps Time: 0.014999881386756897
Found 402.038671875 in 37 steps Time: 0
Found 402.238671875 in 38 steps Time: 0.019999686628580093
Found 412.5953125 in 34 steps Time: 0
Found 412.7953125 in 32 steps Time: 0.005000270903110504
Found 422.5953125 in 36 steps Time: 0
Found 422.7953125 in 33 steps Time: 0
Found 437.5953125 in 34 steps Time: 0.004999805241823196
Found 437.7953125 in 37 steps Time: 0.005000270903110504
Found 457.5953125 in 36 steps Time: 0
Found 457.7953125 in 35 steps Time: 0
Found 476.472265625 in 35 steps Time: 0.005000270903110504
Found 476.672265625 in 34 steps Time: 0.005000270903110504
Found 490.34921875 in 37 steps Time: 0.004999805241823196
Found 490.54921875 in 32 steps Time: 0.0100000761449337
Found 502.009375 in 32 steps Time: 0
Found 502.209375 in 36 steps Time: 0.0100000761449337
Found 514.226171875 in 37 steps Time: 0
Found 514.426171875 in 34 steps Time: 0
Found 529.7828125 in 34 steps Time: 0.004999805241823196
Found 529.9828125 in 37 steps Time: 0.005000270903110504
Found 549.7828125 in 36 steps Time: 0.015000347048044205
Found 549.9828125 in 37 steps Time: 0.009999610483646393
Found 564.7828125 in 37 steps Time: 0.004999805241823196
Found 564.9828125 in 37 steps Time: 0.005000270903110504
Found 575.339453125 in 37 steps Time: 0.005000270903110504
Found 575.539453125 in 38 steps Time: 0.004999805241823196
Found 590.89609375 in 34 steps Time: 0
Found 591.09609375 in 35 steps Time: 0
Found 605.89609375 in 36 steps Time: 0
Found 606.09609375 in 34 steps Time: 0.004999805241823196
Found 617.55625 in 38 steps Time: 0
Found 617.75625 in 39 steps Time: 0.0100000761449337
Found 629.773046875 in 39 steps Time: 0
Found 629.973046875 in 37 steps Time: 0
Found 645.3296875 in 38 steps Time: 0
Found 645.5296875 in 38 steps Time: 0
Found 665.3296875 in 40 steps Time: 0
Found 665.5296875 in 38 steps Time: 0.005000270903110504